### PR TITLE
Added functionality to generate NPQ certificates

### PR DESCRIPTION
**Pull request by hortha**
### Context

Need to be able to generate an NPQ Certificate in the qualifications v3 API endpoint to be callable from the new UI

### Changes proposed in this pull request

Added new certificates/npq endpoint and supporting code / tests.

### Guidance to review

As well as adding support for NPQ certificates, refactored QTS to use shared code + associated tests.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[NPQ certificates endpoint](https://trello.com/c/safYgMwD/788-npq-certificates-endpoint)

File: QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetNpqCertificateHandler.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        var teacher = qualification.Extract<Contact>(Contact.EntityLogicalName, Contact.PrimaryIdAttribute);
+        if (teacher == null)
+        {
+            return null;
+        }
+
+        var fieldValues = new Dictionary<string, string>()
+        {
```
Comment (gunndabad) [inline QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetNpqCertificateHandler.cs:63]: Nit: double space if middle name is empty?

**Review by gunndabad — APPROVED** [review]

File: QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetNpqCertificateHandler.cs (hunk 1 above)
Comment (hortha) [reply QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetNpqCertificateHandler.cs:63]: Interestingly it is the same as currently in the old system!
I'll fix it though as it is not hard

Comment (hortha) [reply QualifiedTeachersApi/src/QualifiedTeachersApi/V3/Handlers/GetNpqCertificateHandler.cs:73]: done

**Pull request merged**